src/sqlfluff/core/rules/std.py

```diff
@@ -3462,11 +3462,14 @@ def _get_indexes(segment):
                 cnt_select_targets += 1
                 if first_select_target_idx == -1:
                     first_select_target_idx = fname_idx
-            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx - 1:
+            if seg.is_type("keyword") and seg.name == "SELECT" and select_idx == -1:
                 select_idx = fname_idx
             if seg.is_type("newline") and first_new_line_idx == -1:
                 first_new_line_idx = fname_idx
-            if seg.is_type("whitespace") and first_whitespace_idx == -1:
+            # TRICKY: Ignore whitespace prior to the first newline, e.g. if
+            # the line with "SELECT" (before any selct targets) has trailing
+            # whitespace.
+            if seg.is_type("whitespace") and first_new_line_idx != -1 and first_whitespace_idx == -1:
```

Comment:
This is the real fix. By ignoring whitespace prior to the first newline, we avoid being fooled by trailing whitespace after the `SELECT`.

Question: Does this potentially cause other issues?